components/boards/Playboard.js

```diff
@@ -23,57 +30,111 @@ function hexagonIndexIsInPath(winningPath, index) {
   return _.indexOf(winningPath, (index + 1).toString(10)) >= 0;
 }
 
-function Playboard(props) {
-  const size = props.size;
+/**
+ * Initialize states.
+ *
+ * @param {int} size
+ */
+function init(size) {
+  return {
+    winner: NO_PLAYER_VALUE,
+    player: FIRST_PLAYER_VALUE,
+    grid: generateEmptyGrid(size),
+  };
+}
 
-  const boardRatio = getBoardRatio(size);
+/**
+ * Play a move based on the id of hexagon clicked.
+ * Updates player, grid and winner values.
+ *
+ * @param {int} id
+ * @param {*} state
+ */
+function playMove(id, state) {
+  if (state.grid[id] !== 0 || state.winner) {
+    return { player: state.player, grid: state.grid, winner: state.winner };
+  }
 
-  const hexagonWidth = getHexagonWidth(size);
-  const hexagonHeight = getHexagonHeight(size);
+  const updatedGrid = state.grid.map((hexagon, index) =>
+    id === index ? state.player : hexagon
+  );
+
+  const winningPath = getWinningPath(updatedGrid, state.player);
 
-  const [grid, setGrid] = useState(generateEmptyGrid(size));
-  const [player, setPlayer] = useState(FIRST_PLAYER_VALUE);
-  const [winner, setWinner] = useState(0);
+  if (winningPath) {
+    const winningGrid = state.grid.map(function (value, index) {
+      return hexagonIndexIsInPath(winningPath, index)
+        ? WINNER_LINE_VALUE
+        : value;
+    });
 
-  const handleCellOnPress = (id, player) => {
-    if (grid[id] !== 0 || winner) {
-      return;
-    }
-    const updatedGrid = grid.map((hexagon, index) =>
-      id === index ? player : hexagon
-    );
-    setGrid(updatedGrid);
+    return { player: state.player, grid: winningGrid, winner: state.player };
+  }
 
-    const winningPath = getWinningPath(updatedGrid, player);
+  const player =
+    state.player === FIRST_PLAYER_VALUE
+      ? SECOND_PLAYER_VALUE
+      : FIRST_PLAYER_VALUE;
 
-    if (winningPath) {
-      setWinner(player);
+  return { player, grid: updatedGrid, winner: state.winner };
+}
 
-      const winningGrid = grid.map(function(value, index) {
-        if (hexagonIndexIsInPath(winningPath, index)) {
-          return WINNER_LINE_VALUE;
-        }
+function reducer(state, action) {
+  switch (action.type) {
+    case "reset":
+      return init(action.payload);
+    case "playMove":
+      return playMove(action.payload, state);
+    default:
+      throw new Error();
+  }
+}
 
-        return value;
-      });
+function Playboard({ size }) {
+  const boardRatio = getBoardRatio(size);
 
-      setGrid(winningGrid);
-    }
+  const hexagonWidth = getHexagonWidth(size);
+  const hexagonHeight = getHexagonHeight(size);
+
+  const [state, dispatch] = useReducer(reducer, size, init);
```

Comment:
you could destructure here like this:
```suggestion
  const [{ grid, player, winner }, dispatch] = useReducer(reducer, size, init);
```

Reply:
Fixed, thanks for the tip !